spec.bs

```diff
@@ -464,23 +516,63 @@ To <dfn>create a request</dfn> given a [=URL=] |url|, and a [=string=] <dfn for=
-  "us-ascii", return failure.
-1. Return |body|.
+  set to |mimeType|.
+1. Let |resource| be null.
+1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given a
+  [=response=] |response| and |responseBody|:
+  1. If |responseBody| is null or failure, set |resource| to failure.
+  1. Let |headers| be |response|'s [=response/header list=].
+  1. If [=header list/getting a structured field value=] "X-Allow-FLEDGE" from |headers| does not
+    return true, set |resource| to failure.
+  1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
+  1. Set |resource| to failure. if any of:
+    1. |headerMimeType| is failure.
+    1. |mimeType| is "`text/javascript`" and |headerMimeType| is not a [=JavaScript MIME type=].
+    1. |mimeType| is "`application/json`" and |headerMimeType| is not a [=JSON MIME type=].
+  1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].
+  1. Set |resource| to failure. if any of:
+    1. If |mimeTypeCharset| does not [=map/exist=], or |mimeTypeCharset| is "utf-8", and
+      |responseBody| is not [=UTF-8=] encoded.
+    1. If |mimeTypeCharset| is "us-ascii", and |responseBody| is not [=ascii string=].
+1. Wait for |resource| to be set.
+1. Return |resource|.
+
+</div>
+
+<div algorithm="build trusted bidding signals url">
+
+To <dfn>build trusted bidding signals url</dfn> given a [=URL=] |signalsUrl|, an [=ordered set=] of
+[=strings=] |keys|, and an [=ordered set=] of [=strings=] |igNames|:
+1. Let |queryParamsList| be a new [=list/is empty|empty=] [=list=].
+1. [=list/Append=] "hostname=" to |queryParamsList|.
+1. [=list/Append=] the result of [=UTF-8 percent encode=] [=this=]'s [=relevant settings object=]'s
+  [=environment/top-level origin=] to |queryParamsList|.
+1. If |keys| is not [=set/is empty|empty=]:
+  1. [=list/Append=] "&keys=" to |queryParamsList|.
+  1. Let |keysStr| be the result of [=string/concatenating=] |keys| with separator set to ",".
+  1. [=list/Append=] the result of [=UTF-8 percent encode=] |keysStr| to |queryParamsList|.
+1. If |igNames| is not [=set/is empty|empty]:
+  1. [=list/Append=] "&interestGroupNames=" to |queryParamsList|.
+  1. Let |igNamesStr| be the result of [=string/concatenate=] |igNames| with separator set to ",".
+  1. [=list/Append=] the result of [=UTF-8 percent encode=] |igNamesStr| to |queryParamsList|.
+1. TODO: Do we want to add [=auction config/seller experiment group id=]?
+1. Let |fullSignalsUrl| be |signalsUrl|.
+1. Set |fullSignalsUrl|'s [=url/query=] to the result of [=string/concatenating=] |queryParamsList|.
+1. return |fullSignalsUrl|.
+
+</div>
+
+<div algorithm="UTF-8 percent encode">
+
+To <dfn>UTF-8 percent encode</dfn> a [=scalar value string=] |input|, return the result of
+[=string/UTF-8 percent-encoding=] |input| using <code>application/x-www-form-urlencoded</code>
+percent-encode set, and boolean spaceAsPlus set to true.

```

Comment:
we're thinking about escaping spaces using %20 instead of "+" to be consistent with the standard. So writing the spec using %20. Will come back to this if we decide to keep "+".